src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -438,7 +436,7 @@ Shard::storeLedger(
     if (!batch.empty() && !storeBatch())
         return fail("Failed to store");
 
-    return std::make_tuple(storeCount, storeSize, false);
+    return result;
```

Comment:
If `StoreLedgerResult::error` self-initialized to `true` then this is the only place you would need to explicitly change that value.

Reply:
You would need to also change the value to `false` in the first two checks.